File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters( string text );
int count_words( string text );
int count_sentences( string text );

int main(void)
{
    string text = get_string("Text: ");
    float L = count_letters( text ) , S = count_sentences( text) ;
    int W = count_words( text ) ;

    L = (L * 100) / W ;
    S = (S * 100) / W ;

    float index = ((0.0588 * L) - (0.296 * S) - 15.8) ;
    index = roundf( index );

    if (index >= 16)
    {
        printf ("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf ("Before Grade 1\n");
    }
    else
    {
        printf ("Grade %.0f\n" , index);
    }
}


int count_letters( string text )
{
    int count = 0;
    for(int i = 0 , n = strlen(text); i < n ; i++)
    {
        if(isalpha(text[i]))
        {
            count++;
        }
    }
    return count;
}


int count_words( string text )
{
    int count = 0;
    for(int i = 0 , n = strlen(text); i < n ; i++)
    {
        if(text[i]==' ')
        {
            count++;
        }
    }
    return count+1;
}


int count_sentences( string text )
{
    int count = 0;
    for(int i = 0 , n = strlen(text); i < n ; i++)
    {
        if(text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            count++;
        }
    }
    return count;
}

File: readability.py
from cs50 import get_string


# main function
def main():
    # Get the text from user
    text = get_string("Text: ")

    # Get the stats of the input text
    L = countletters(text)
    W = countwords(text)
    S = countsentences(text)

    L = (L * 100.0) / W
    S = (S * 100.0) / W

    # Calculate the Coleman-Liau Value according to input text
    index = round(0.0588 * L - 0.296 * S - 15.8)

    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


# Function to count the number of letters in the text
def countletters(text):
    count = 0
    for i in text:
        if i.isalpha():
            count += 1
    return count


# Function to count the number of words in the text
def countwords(text):
    return len(text.split())


# Function to count the number of sentences in the text
def countsentences(text):
    count = 0
    for i in text:
        if i == '.' or i == '!' or i == '?':
            count += 1
    return count


main()